pkg/services/rpcsrv/server_test.go

```diff
@@ -3350,3 +3504,43 @@ func TestFailedPreconditionShutdown(t *testing.T) {
 
 	require.Eventually(t, stopped.Load, 5*time.Second, 100*time.Millisecond, "Shutdown should return")
 }
+
+func testFunctionsWithOldStates(t *testing.T, doRPCCall func(string, string, *testing.T) []byte) {
+	chain, rpcSrv, httpSrv := initClearServerWithCustomConfig(t, func(c *config.Config) {
+		c.ApplicationConfiguration.Ledger.KeepOnlyLatestState = true
+	})
+
+	//chain, rpcSrv, httpSrv := initServerWithInMemoryChain(t)
+
+	defer chain.Close()
+	defer rpcSrv.Shutdown()
+
```

Comment:
And what you can do is instead of creating function that looks exactly like `testRPCProtocol`, but have a special blockchain instance running for it, try to move blockchain instance creation to a separate function and pass ready-to-use blockchain to `testRPCProtocol`. This allows you to avoid massive code duplication between `testRPCProtocol` and `testFunctionsWithOldStates`. You can also pass `rpcFunctionsWithOldStatesTestCases` or `rpcTestCases` as a parameter to that function.

Reply:
Completely agree. It was just because I pushed it as it was. I started testing without any idea how many changes it needs. Fixed.